lib/ansible/plugins/connection/__init__.py

```diff
@@ -123,17 +137,17 @@ def _split_ssh_args(argstring):
 
     @property
     @abstractmethod
-    def transport(self):
+    def transport(self) -> str:
         """String used to identify this Connection class from other classes"""
         pass
 
     @abstractmethod
-    def _connect(self):
+    def _connect(self) -> ConnectionBase:
```

Comment:
debating if this should be something like:
```suggestion
    def _connect(self: T) -> T:
```
with T maybe being a bound typevar to `ConnectionBase`?

Mostly academic exercise in this case anyway, I suppose, unless we actually wanted to allow a connection to return a different type for some reason :laughing: